tests/MongoDB.Driver.Tests/Specifications/unified-test-format/UnifiedErrorMatcher.cs

```diff
@@ -0,0 +1,100 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Linq;
+using FluentAssertions;
+using MongoDB.Bson;
+
+namespace MongoDB.Driver.Tests.Specifications.unified_test_format
+{
+    public class UnifiedErrorMatcher
+    {
+        private UnifiedValueMatcher _valueMatcher;
+
+        public UnifiedErrorMatcher(UnifiedValueMatcher valueMatcher)
+        {
+            _valueMatcher = valueMatcher;
+        }
+
+        public void AssertErrorsMatch(BsonDocument expectedError, Exception actualException)
+        {
+            foreach (var element in expectedError)
+            {
+                switch (element.Name)
+                {
+                    case "isError":
+                        var isError = element.Value.AsBoolean;
+                        isError.Should().BeTrue("Test files MUST NOT specify false.");
+                        actualException.Should().NotBeNull();
+                        break;
+                    case "isClientError":
+                        var isClientError = element.Value.AsBoolean;
+                        // TODO: Recheck assertion and add types as necessary.
+                        var actualIsClientError =
+                            actualException is MongoClientException ||
+                            actualException is BsonException;
+                        actualIsClientError.Should().Be(isClientError);
+                        break;
+                    case "errorContains":
+                        var expectedSubstring = element.Value.AsString;
+                        var actualExceptionContains = actualException.Message.IndexOf(expectedSubstring, StringComparison.OrdinalIgnoreCase) >= 0;
+                        actualExceptionContains.Should().BeTrue();
```

Comment:
Is it because the JSON files have the wrong case for some error messages? If so consider:

    actualException.Message.Should().ContainEquivalentOf(expectedSubstring);

But you need to verify that `ContainEquivalentOf` really is a case-insensitive version of `Contain`.